### PR TITLE
update numba pinning

### PR DESCRIPTION
**Pull request by quasiben**
PR updates the numba pinning for dask-cuda.  We exclude `0.51.0` due to a small bug related to cuda context creation.  PR https://github.com/numba/numba/pull/6147 address the problem and `0.51.1` will be released in the future

cc @jakirkham @kkraus14 @pentschev

### CODE REVIEW

**Review by pentschev — APPROVED** [review]
Thanks @quasiben .

**Comment by quasiben** [comment]
Let's wait for @jakirkham before merging

**Comment by pentschev** [comment]
> Let's wait for @jakirkham before merging

I think we'll need @rapidsai/admins to merge this.

**Comment by kkraus14** [comment]
I have the power to merge, but will wait for @jakirkham's feedback.

**Comment by jakirkham** [comment]
Thanks all! 😀

**Comment by pentschev** [comment]
> I have the power to merge, but will wait for @jakirkham's feedback.

No need to show off @kkraus14 😆

**Pull request merged**
